A fix for a crash when uploading media on certain devices

planarvoid opened the pull request:
Fixes https://github.com/wordpress-mobile/WordPress-Android/issues/8978
This PR should fix a crash that I couldn't reproduce. However, googling the issue led me to [this solution](https://stackoverflow.com/a/53021624) to the same problem. I'm using methods from the [FileUtils](https://github.com/coltoscosmin/FileUtils/blob/master/FileUtils.java) with some modifications that should prevent this problem. 

This solution should cover all 3 possible file destinations ("content://downloads/public_downloads", "content://downloads/my_downloads", "content://downloads/all_downloads") + should fall back to loading the file into memory with a stream when everything else fails. 

@daniloercoli I'll ping you once I create a PR for the WPAndroid to test this change

To test:
* I couldn't reproduce this issue so I don't really know how to test it.

daniloercoli commented on libs/utils/WordPressUtils/src/main/java/org/wordpress/android/util/FileUtils.java:
```diff
@@ -58,4 +73,127 @@ public static String getFileNameFromPath(String filePath) {
+    }
+
+    @Nullable
+    private static File generateFileName(@Nullable String name, File directory) {
```
Wondering it we can re-use the already available routine [MediaUtils.getUniqueCacheFileForName ](https://github.com/wordpress-mobile/WordPress-Android/blob/bb8f7ee2d4d02c960b6e76beb14f6bedf4d1fc68/libs/utils/WordPressUtils/src/main/java/org/wordpress/android/util/MediaUtils.java#L241) ?

planarvoid replied:
I think we can! good point 👍

daniloercoli commented on libs/utils/WordPressUtils/src/main/java/org/wordpress/android/util/FileUtils.java:
```diff
@@ -58,4 +73,127 @@ public static String getFileNameFromPath(String filePath) {
+     */
+    static String cacheFile(Context context, Uri uri) {
+        String fileName = getFileName(context, uri);
+        File cacheDir = getDocumentCacheDir(context);
```
Why not reusing the application cache dir here?  `File cacheDir = context.getCacheDir();` Pretty much what we use in MediaUtils [here](https://github.com/wordpress-mobile/WordPress-Android/blob/bb8f7ee2d4d02c960b6e76beb14f6bedf4d1fc68/libs/utils/WordPressUtils/src/main/java/org/wordpress/android/util/MediaUtils.java#L196).

planarvoid replied:
I think we can. This is creating a nested dir which (I think) is not necessary)

planarvoid commented:
@daniloercoli thanks for the tip! I think you're right. I've replaced the new methods with the method you suggested and removed the code that's now unused. I couldn't reproduce the crash even before. I think it only happens on some devices.

daniloercoli commented:
I was able to reproduce the problem on a Nexus5X on a Android 7.1 ! 🎉 

- Go to Media Library
- Upload a picture already on the device that was previously downloaded from Internet in the `Download` folder.
- Boom!


This PR does fix the issue! 👍
